parastell/utils.py

```diff
@@ -311,3 +318,74 @@ def combine_dagmc_models(models_to_merge):
             renumberizer.load_file(temp_filename)
     renumberizer.renumber_ids()
     return dagmc.DAGModel(renumberizer.mb)
+
+
+def stl_to_cq_solid(stl_path, tolerance=0.001):
+    """Create a solid volume in CadQuery from an STL file containing the
+    triangles defining a watertight DAGMC volume. The resulting volume should
+    have surfaces that exactly match the STL files provided. Useful for
+    obtaining CAD representations of volumes in DAGMC models.
+
+    Arguments:
+        stl_path (str): Path to the STL file defining the volume.
+        tolerance (float): Distance (in whatever units the STL file is in)
+            at which to consider vertices coincident or edges colinear.
+
+    Returns:
+        cq_solid (CadQuery Solid): Solid volume with surfaces defined by the
+            the triangles in the provided STL files.
+    """
+    sewer = BRepBuilderAPI_Sewing(tolerance)
+
+    reader = StlAPI_Reader()
+    shape = TopoDS_Shape()
+    reader.Read(shape, stl_path)
+    sewer.Add(shape)
+
+    sewer.Perform()
+    sewn_shape = sewer.SewedShape()
+
+    shell_explorer = TopExp_Explorer(sewn_shape, TopAbs_SHELL)
+    shell_shape = shell_explorer.Current()
+
+    cq_solid = cq.Solid.makeSolid(cq.Shape(shell_shape))
+
+    return cq_solid
+
+
+def dagmc_volume_to_step(
+    dagmc_model, volume_id, step_file_path, tolerance=0.001
+):
+    """Create a STEP file with surfaces defined by the triangles making up the
+    surface of a DAGMC volume and save it to file.
+
+    Arguments:
+        dagmc_model (PyDAGMC DAGModel object): DAGMC model containing the
+            volume to be converted to STEP.
+        volume_id (int): ID of the volume to be converted to STEP.
+        step_file_path (str): Path on which to save the resulting step file.
+        tolerance (float): Distance (in whatever units the STL file is in)
+            at which to consider vertices coincident or edges colinear.
+            Defaults to 0.001.
+    """
+    volume = dagmc_model.volumes_by_id[volume_id]
+
+    num_tris = len(volume.triangle_handles)
+
+    with tempfile.NamedTemporaryFile(delete=True, suffix=".stl") as temp_file:
+        stl_path = temp_file.name
+        volume.model.mb.write_file(
```

Comment:
Since you already have the model, this may be better
```suggestion
        dagmc_model.mb.write_file(
```